imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/Image.java

```diff
@@ -1,29 +1,23 @@
 package pl.edu.uj.kimage;
 
-//only for RGBA
-public class Image {
-    private int[] data;
-    final private int width;
-    final private int height;
+public final class Image {
+    private final Color[] data;
+    private final int width;
+    private final int height;
 
     public Image() {
         width = height = 0;
         data = null;
     }
 
-    public Image(final int width, final int height) {
+    public Image(final int width, final int height, final Color[] data) {
+        if(data.length != width * height) {
+            throw new IllegalArgumentException("Size of image does not match with input color data size.");
```

Comment:
`Add message to each exception with explanation of cause with all additional data. For example when the file was not found add path to this file in exception message.` from `CONTRIBUTING.md`